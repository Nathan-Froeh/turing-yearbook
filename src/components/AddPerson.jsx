import React, {Component} from 'react'
import './AddPerson.scss'

class AddPerson extends Component {
constructor(props) {
  super(props)
  this.state = {
    name: '',
    quote: '',
    superlative: ''
  }
}

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleAdd = (e) => {
    e.preventDefault()
    const {name, quote, superlative} = this.state
    this.props.addStudent(name, quote, superlative)
    this.props.hide()
  }

  render() {
    return(
      <section 
        className='add-box' 
          onSubmit={this.handleAdd}>
        <form action='add person'>
          <label htmlFor="name" >Name</label>
          <input 
           type='text' 
            name='name' 
            id='name' 
            onChange={this.handleChange}
          />
          <label htmlFor="quote">Quote</label>
          <input 
            type="text" 
            name='quote' 
            id='quote' 
            onChange={this.handleChange}
          />
          <label htmlFor="superlative">Superlative</label>
          <input 
            type="text" 
            name='superlative' 
            id='superlative' 
            onChange={this.handleChange}
          />
          <input type="submit" value='Submit'/>
        </form>
      </section>
    )
  }
}

export default AddPerson;