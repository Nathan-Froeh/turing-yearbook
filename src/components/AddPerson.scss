.add-box {
  height: 100vh;
  width: 100vw;
  position: fixed;
  z-index: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  background: rgba(0,0,0,.8);
  form {
    height: 400px;
    width: 400px;
    background: white;
  }
}

.hidden {
  display: none;
}