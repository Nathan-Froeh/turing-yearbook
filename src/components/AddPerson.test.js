import React from 'react';
import AddPerson from './AddPerson';
import {shallow} from 'enzyme';


describe('AddPerson', () => {

  it('should render all components in order', () => {
    const wrapper = shallow(
      <AddPerson addStudent={jest.fn()} hide={jest.fn()}/>
    )
    expect(wrapper).toMatchSnapshot()
  })


}) 