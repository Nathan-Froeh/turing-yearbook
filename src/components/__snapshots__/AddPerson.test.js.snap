// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPerson should render all components in order 1`] = `ShallowWrapper {}`;
